eos-tech-support/eos-write-live

```diff
@@ -0,0 +1,272 @@
+    EOS_DOWNLOAD_IMAGE='eos-download-image'
+fi
+
+ARGS=$(getopt -o o:x:,lnmwfh -l "os-image:,windows-tool:,latest,ntfs,mbr,writable,force,help" -n "$0" -- "$@")
+eval set -- "$ARGS"
+
+usage() {
+    cat <<EOF
+Usage:
+   $0 [options] DEVICE
+
+Arguments:
+    DEVICE                 Device path (e.g. '/dev/sdb')
+
+Options:
+   -o,--os-image PATH      Path to Endless OS image
+   -x,--windows-tool PATH  Path to Endless OS installer tool for Windows
+   -l,--latest             Fetch latest OS image and/or Windows tool
+   -w,--writable           Allow image to be modified when run (which
+                           prevents installing it later)
+   -f,--force              don't ask to proceed before writing
+   -h,--help               Show this message
+
+Developer options (you probably don't want to use these):
+   -n,--ntfs               Format the image partition as NTFS, not exFAT
+   -m,--mbr                Format DEVICE as MBR, not GPT (implies --ntfs)
+EOF
+}
+
+function check_exists() {
+    if [ ! -f "$1" ]; then
+        echo "$2 $1 does not exist or is not a file" >&2
+        exit 1
+    fi
+}
+
+NTFS=false
+MBR=false
+WRITABLE=false
+FORCE=false
+while true; do
+    case "$1" in
+        -o|--os-image)
+            shift
+            OS_IMAGE="$1"
+            shift
+            ;;
+        -x|--windows-tool)
+            shift
+            WINDOWS_TOOL="$1"
+            shift
+            ;;
+        -l|--latest)
+            shift
+            FETCH_LATEST=true
+            ;;
+        -n|--ntfs)
+            shift
+            NTFS=true
+            ;;
+        -m|--mbr)
+            shift
+            MBR=true
+            NTFS=true
+            ;;
+        -w|--writable)
+            shift
+            WRITABLE=true
+            ;;
+        -f|--force)
+            shift
+            FORCE=true
+            ;;
+        -h|--help)
+            usage
+            exit 0
+            ;;
+        --)
+            shift
+            break
+            ;;
+    esac
+done
+
+if [ $# -lt 1 ] ; then
+    echo "Missing DEVICE" >&2
+    usage >&2
+    exit 1
+fi
+
+DEVICE="$1"
+
+USERID=$(id -u)
+if [ "$USERID" != "0" ]; then
+    echo "Program requires superuser privileges" >&2
+    exit 1
+fi
+
+# Check for required tools
+if $NTFS; then
+    MKFS_IMAGES="mkfs.ntfs"
+    MKFS_ARGS='--quick -L'
+    MKFS_PACKAGE=ntfs-3g
+else
+    MKFS_IMAGES="mkfs.exfat"
+    MKFS_ARGS=-n
+    MKFS_PACKAGE=exfat-utils
+fi
+
+command -v $MKFS_IMAGES >/dev/null || {
+    echo "$MKFS_IMAGES is not installed... aborting!" >&2
+    echo "Try 'sudo apt-get install $MKFS_PACKAGE'" >&2
+    exit 1
+}
+
+command -v dd >/dev/null || {
+    echo "dd is not installed... aborting!" >&2
+    echo "Try 'sudo apt-get install coreutils'" >&2
+    exit 1
+}
+
+command -v unzip >/dev/null || {
+    echo "unzip is not installed... aborting!" >&2
+    echo "Try 'sudo apt-get install unzip'" >&2
+    exit 1
+}
+
+if test \! -b $DEVICE; then
```

Comment:
I think inside a script you can skip escaping `!` since `histexpand` is off in non-interactive shells. Doesn't hurt, though.